Change Note message to correspond to phoenix-core Note

jules opened the pull request:
This also fixes a typo in the encrypted_data
field.

Fixes #36

jules committed:
Change Note message to correspond to phoenix-core Note

This also fixes a typo in the encrypted_data
field.

Fixes #36

CPerezz commented on transaction.proto:
```diff
@@ -15,12 +16,14 @@ message Fee {
-    JubJubCompressed commitment = 3;
-    BlsScalar nonce = 4;
-    PoseidonCipher encypted_data = 5;
+    int32 note_type = 1;
```
Not sure about that. We unified them in phoenix_core, so why do we specify ghe type here? Can you check it with @ZER0 ?

jules replied:
The struct in `phoenix-core` still has this identifier too though. I reckon we should keep it, otherwise we would need to infer based on the underlying data whether or not a note is obfuscated/transparent, and that sounds like a needless hassle. I'm gonna guess this is the same reason that the identifier still exists in `phoenix-core`, but I'll let @ZER0 weigh in on that.

ZER0 replied:
@CPerezz exactly because they're unified we need to specify the type there, otherwise you will have two different messages, `TransparentNote` and `ObfuscatedNote`. Since, instead, we have one single `Note` we need a field to indicate the type.

However I would prefer using an `enum` in the schema:
```
message Note {
    enum NoteType {
        TRANSPARENT = 0;
        OBFUSCATED = 1;
    }
    NoteType note_type = 1;
   // etc
}

CPerezz reviewed (commented):
Just this question/topic. Otherways, LGTM.

ZER0 commented on transaction.proto:
```diff
@@ -15,12 +16,14 @@ message Fee {
     JubJubCompressed R = 3;
     JubJubCompressed pk_r = 4;
```
Remember that we don't need this: every time you have the couple `R` + `pk_r` means you basically have a `StealthAddress` type. Let's sure we're consistent in that across our schemas.

jules replied:
Good point - I already caught on to this doing some work for the rusk transaction module, but it makes sense to update it here too.

ZER0 commented on transaction.proto:
```diff
@@ -15,12 +16,14 @@ message Fee {
+    int32 note_type = 1;
+    JubJubCompressed value_commitment = 2;
+    JubJubScalar nonce = 3;
+    StealthAddress pk_r = 4;
```
as already mentioned in a previous review, `StealthAddress` and `pk_r` are two different thing: `pk_r` is a part of `StealthAddress`. So the name would be `stealth_address`, or `addr` or whatever you like, but not `pk_r` since would be misleading (you will end up with `note.pk_r.pk_r` structure)

jules replied:
Yep, gotcha

jules committed:
Address review comments

This entails:
- Unifying the `r` and `pk_r` fields in the Fee message
- Renaming the StealthAddress in the Note to `address`
- Adding an enum for the note type

CPerezz commented:
@ZER0 I thought that since the fields of the protobuf are `Option<T>` we could derive the note type from there. But that's also fine for me.

CPerezz reviewed (approved):
With all of @ZER0's comments addressed. LGTM
